org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/Dialog.java

```diff
@@ -387,7 +387,15 @@ public static void askForInput(int windowNo, String adMessage, final Callback<St
     	askForInput(windowNo, adMessage, "", callback);
     }
     
+    public static void askForSecretInput(int windowNo, String adMessage, final Callback<String> callback) {
+    	askForInput(windowNo, adMessage, "", callback, true);
```

Comment:
I think it is much cleaner that we just call the version of showDialog that takes editor and pass it the wpasswordeditor. The new showdiaog method looks a bit strange now as the isInputProtected parameter is just ignore if editor is not null. The additional isInputProtected parameter looks unnecessary too since the method accept editor parameter and you can just pass it a wpasswordeditor and has full control over it.

Reply:
You're right. I tried that at first and WPasswordEditor didn't have a usable constructor, I had to add it anyway so closing the pull request to create a new one with the requested changes